tests/test_other.py

```diff
@@ -9443,3 +9443,16 @@ def test_EMSCRIPTEN_and_STRICT(self):
     ''')
     run_process([PYTHON, EMCC, 'src.c', '-c'])
     self.expect_fail([PYTHON, EMCC, 'src.c', '-s', 'STRICT', '-c'])
+
+  def test_exceptions_modes_symbols_defined(self):
+    for catching in [0, 1]:
+      for throwing in [0, 1]:
+        for opts in [0, 1]:
+          cmd = [PYTHON, EMCC, path_from_root('tests', 'other', 'exceptions_modes_symbols_defined.cpp'), '-s', 'DISABLE_EXCEPTION_THROWING=%d' % (1 - throwing), '-s', 'DISABLE_EXCEPTION_CATCHING=%d' % (1 - catching), '-O%d' % opts]
```

Comment:
This style of unit testing with loops gives be a bad feeling.   In this situation I can understand any you did it.  Maybe at least this `@parametarised` based on opts to take it from a tripple to double nested loop?

Also, maybe put this in `test_core.py` so that it gets tested in different configurations automatically?

Also,  perhaps we can come up with a more descriptive name for this test?   How about "test_exception_settings"?

Reply:
Hmm, `@parameterized` requires a name for each test, so in such cartesian products it gets burdensome, unless there's some trick?

I changed it to use `itertools.product` which is shorter.

This feels excessive to me in `core` - is there a specific reason you think we need more coverage? I guess I'm assuming linking doesn't change much between modes, which maybe is too much to assume.

Btw I'm curious why it gives you a bad feeling?

Reply:
Landing as this may help users currently getting confusing errors, but happy to improve the tests more if people want later.

Reply:
Hypothetically, you *could* do:

```python
@parameterized({'_'.join(name for name, enable in zip(['catching', 'throwing', 'opts'], flags) if enable): flags for flags in itertools.product([0, 1], repeat=3)})
def test_exception_settings(self, catching, throwing, opts):
    ...
```

But I think this is a little excessive.

Reply:
That's cool :) but yeah, kind of complicated...